src/generator.js

```diff
@@ -33,11 +41,35 @@ function createPHPFile(folder) {
             return
         }
 
+        const category = detectCategory(name);
+
+        if (category == 'class') {
+            return window.showQuickPick([
+                "Class",
+                "Trait",
+                "Interface"
+            ], {
+                title: "Choose class type",
+                canPickMany: false,
+                matchOnDescription: true,
+                matchOnDetail: true
+            });
+        }
+
+        return category
```

Comment:
why `return category` ?

Reply:
Return category in first then() mean pass this value to the next then().
Or if return quick pick's promise, mean wait this promise and pass promise value to next then().

Reply:
If don't use another then(), we need two duplicated in showQuickPick and else code block to call createNewFile(), that's ugly.